subprojects/shipkit/src/test/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFileTest.groovy

```diff
@@ -12,17 +12,30 @@ class CreateDependencyInfoFileTest extends PluginSpecification {
 
         def child = new ProjectBuilder().withName("child").withParent(project).build()
         child.group = "projGroup"
-        def sibling = new ProjectBuilder().withName("sibling").withParent(project).build()
+        def sibling = new ProjectBuilder().withName("submodule").withParent(project).build()
         sibling.version = "1.2.3"
         sibling.group = "projGroup"
 
         child.repositories.jcenter()
         child.configurations.create("runtime")
         child.dependencies {
-            runtime child.project(":sibling")
+            runtime child.project(":submodule") // current version of submodule -> skip version info
+            runtime "projGroup:submodule2:1.0.0" // old version of submodule2 -> keep version, it's not the same as child project's version
             runtime "junit:junit:4.10"
-            runtime "org.mockito:mockito-core:2.10.0:sources"
-            runtime "org.mockito:mockito-core:2.10.0"
+            runtime (group: 'org.mockito', name: 'mockito-core', version: '2.10.0') {
+                artifact {
+                    name = 'mockito'
+                    extension = 'jar'
+                    type = 'sources'
+                    classifier = 'jar'
+                }
+                artifact {
```

Comment:
Nice! Thank you for adding coverage for that!

Can you spice up the test data and change/add artifact that does not declare all fields? (e.g. has nulls)